### ISSUE TITLE
Add error code to VolumeSnapshotError

### ISSUE THREAD

**Issue by ggriffiths** [kubernetes-csi/external-snapshotter#431]
Would it be possible to add a VolumeSnaphotError.Code field to classify different error messages and types?
https://github.com/kubernetes-csi/external-snapshotter/blob/8751eb0ca5b33a8a3c2589d506b48cc26684e0cd/client/apis/volumesnapshot/v1/types.go#L429

For example, if a backup system is checking the VolumeSnapshot object status, it would need to check the error message for a string that might change. Since the API must be backwards compatible, having an error code would be more dependable. 

Examples usage of this error code could be here: 
https://github.com/kubernetes-csi/external-snapshotter/blob/0a5839eb4ce39dd36e22c348e538cfcc78657f03/pkg/snapshotter/snapshotter.go#L137-L139
If a snapshotID does not exist in the storage backend, external systems looking at the Snapshot.Status could tell from the VolumeSnapshot.Status.Error.Code. 

The implementation might involve returning VolumeSnapshotError deeper in the call stack at the original origin of the error, but this might also help with classifying different snapshot-controller error metrics to report. 

Another example might be to classify invalid arguments or system errors.

**Comment by xing-yang** [comment]
It will be great if we can come up with error codes that everyone can agree to, but it is not trivial to determine error codes.  CSI spec only uses existing gRPC error codes.  We can try to use existing gRPC error codes when appropriate as well.  For snapshot id not found, I think we should be able to use gRPC NOT_FOUND error code.

We also tried to come up with common error codes when working on volume health, but at the end we decided to start with a boolean `abnormal`.
